dashboard/app/models/email_preference.rb

```diff
@@ -22,14 +22,25 @@ class EmailPreference < ApplicationRecord
   validates_uniqueness_of :email
   validates_email_format_of :email
 
+  SOURCE_TYPES = [
+    REGISTRATION = 'Teacher account sign up',
```

Comment:
I wonder if these should be more like `"teacher_account_sign_up"` so they're a little more formal and less organic, for querying purposes?

Reply:
We will need to loc all such strings, so being humanly readable in code isn't important from that point of view.